### PR TITLE
fix: add retry back off for T04 errors

### PR DESCRIPTION
**Pull request by traviscrist**
Add back the logic to increase the retry time for T04 errors.
Add error message to indicate the issue originates from the ilp-protocol-stream module.

### CODE REVIEW

File: src/connection.ts [hunk 1]
```diff
@@ -953,8 +953,9 @@ export class Connection extends EventEmitter {
             newTestMax = this.maximumPacketAmount.plus(this.testMaximumPacketAmount).dividedToIntegerBy(2)
             this.debug(`maximum packet amount is between ${this.testMaximumPacketAmount} and ${this.maximumPacketAmount}, trying: ${newTestMax}`)
           } else {
-            // Increase by 2 in this case since the amount to send and text max are equal which indicates the last packet was successful
-            newTestMax = this.testMaximumPacketAmount.times(2)
+            // Increase by 10% in this case since the amount to send and text max are equal which indicates the last packet was successful
```
Comment (emschwartz) [inline src/connection.ts]: typo: "te~~x~~**s**t max"

File: src/connection.ts [hunk 2]
```diff
@@ -953,8 +953,9 @@ export class Connection extends EventEmitter {
...
+            const incrementBy = this.testMaximumPacketAmount.times(1.1).integerValue(BigNumber.ROUND_HALF_EVEN)
```
Comment (emschwartz) [inline src/connection.ts]: Is there a reason for making it 1.1? Are you just trying to make it increase a little slower than when it's x2?

Comment (traviscrist) [reply src/connection.ts]: Exactly, no real reason other than 10% being more reasonable than x2

File: src/connection.ts (hunk 1 above)
Comment (traviscrist) [reply src/connection.ts]: updated

**Comment by traviscrist** [comment]
@emschwartz changed the increment logic back to the 2x since otherwise its much too slow. We'll need to revisit this later with the more balanced approach but this should keep the back off with increasing timeout and the ability to increase if we get too low for now.

**Pull request merged**